pkg/monitors/elasticsearch/query/monitor.go

```diff
@@ -0,0 +1,136 @@
+package query
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/signalfx/golib/v3/datapoint"
+	"github.com/signalfx/signalfx-agent/pkg/core/common/httpclient"
+	"github.com/signalfx/signalfx-agent/pkg/core/config"
+	"github.com/signalfx/signalfx-agent/pkg/monitors"
+	"github.com/signalfx/signalfx-agent/pkg/monitors/types"
+	"github.com/signalfx/signalfx-agent/pkg/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig  `yaml:",inline" acceptsEndpoints:"true"`
+	httpclient.HTTPConfig `yaml:",inline"`
+
+	Host string `yaml:"host" validate:"required"`
+	Port string `yaml:"port" validate:"required"`
+
+	MetricQueries []*metricQuery `yaml:"metricQueries" validate:"required"`
+}
+
+// Conceptually the data field in search queries define  bucket aggregations
+// that determine "buckets" (some group of elasticsearch documents that match
+// the search query) and metric aggregations that compute metrics from each "bucket"
+type metricQuery struct {
+	// Index that's being queried
+	Index string `yaml:"index" validate:"required"`
+	// Lucene search query which the elasticsearch documents should match
+	LuceneSearchQuery string `yaml:"luceneSearchQuery" validate:"required"`
```

Comment:
Is this still needed?

Reply:
This is actually redundant. A lucene query string can be provided to match documents on (https://www.elastic.co/guide/en/elasticsearch/reference/current/search-uri-request.html) or looks like this query can be specified in the request body too along with the aggregations (https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-body.html).

Reply:
Removed this config